### PR TITLE
feat: Dock Component

### PR DESCRIPTION
**Pull request by harrymansworth**

### CODE REVIEW

File: src/components/footer-dock-closed/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import DynamicStrapline from "../../presentational/dynamic-strapline/index";
+import DynamicPriceMessaging from "../../presentational/dynamic-price-messaging/index";
+import AnchorLink from "../../presentational/anchor";
+import MarketingImageAsset from "../../presentational/marketing-image-asset";
+import * as TheTimesDeviceSpread from "../../theme/assets/the_times_device_spread.png";
+
+type Props = {
+  articlesRemaining: string;
+  displayRAMessaging: boolean;
+  offsetTop?: string;
```
Comment (hyper0009) [inline src/components/footer-dock-closed/index.tsx]: Prefer `string | null` as `undefined` is a bit vague and I'd prefer if we pass a null value down but this is up to you.

File: src/components/footer-dock-open/index.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  allowanceConsumed: darkTheme
+};
+
+type Props = {
+  articlesRemaining: number;
+  offsetTop?: string;
+  statefulTheme: string;
+};
+
+const FooterOpen: React.FC<Props> = ({
```
Comment (hyper0009) [inline src/components/footer-dock-open/index.tsx]: same as above, we should have defaults.

File: src/presentational/component-hide-button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  ${props =>
+    props.iconRotated
+      ? "transform: rotate(180deg)"
+      : "transform: rotate(0deg)"};
+`;
+
+const Icon = styled.svg`
+  margin-top: 3px;
+  g {
+    fill: ${props => props.theme.fontColors.main}!important;
```
Comment (hyper0009) [inline src/presentational/component-hide-button/index.tsx]: why `important`?

File: src/presentational/remaining-articles-line/index.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    margin-bottom: 10px;
+  }
+`;
+
+type Props = {
+  count: number;
+  string: string;
+};
+
+const RemainingArticles: React.FC<Props> = props => (
```
Comment (hyper0009) [inline src/presentational/remaining-articles-line/index.tsx]: defaults plz

File: src/presentational/button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  @media (max-width: 719px) {
+    max-width: 100%;
+  }
+`;
+
+type Props = {
+  children: string;
+};
+
+const DynamicPriceMessaging: React.FC<Props> = props => (
```
Comment (hyper0009) [inline src/presentational/button/index.tsx]: defaults plz

File: src/presentational/component-hide-button/index.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  }
+`;
+
+type ComponentState = {
+  iconRotated?: boolean;
+  userHidden?: boolean;
+};
+
+const HideButton: React.FC<ComponentState> = props => (
```
Comment (hyper0009) [inline src/presentational/component-hide-button/index.tsx]: defaults plz

File: src/presentational/dynamic-strapline/index.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  @media (max-width: 600px) {
+    max-width: 100%;
+  }
+`;
+
+type Props = {
+  children: string;
+};
+
+const DynamicStrapline: React.FC<Props> = props => (
```
Comment (hyper0009) [inline src/presentational/dynamic-strapline/index.tsx]: more defaults :)

File: src/components/footer-dock-closed/index.tsx (hunk 1 above)
Comment (harrymansworth) [reply src/components/footer-dock-closed/index.tsx]: Decided to remain with `?` pattern

File: src/presentational/component-hide-button/index.tsx (hunk 3 above)
Comment (harrymansworth) [reply src/presentational/component-hide-button/index.tsx]: removed and no visual regression 👍

File: src/presentational/button/index.tsx (hunk 5 above)
Comment (harrymansworth) [reply src/presentational/button/index.tsx]: This component should fail if children property is undefined, so let's not define defaults

File: src/presentational/component-hide-button/index.tsx (hunk 6 above)
Comment (harrymansworth) [reply src/presentational/component-hide-button/index.tsx]: These are 'stateful' properties - if they're undefined that is expected behaviour

File: src/presentational/dynamic-strapline/index.tsx (hunk 7 above)
Comment (harrymansworth) [reply src/presentational/dynamic-strapline/index.tsx]: While this one only uses `children` as it's property we should still define a default for this as it relates to our component messaging priority tree; in short every marketing component should have an agnostic piece of messaging. This is yet to be defined and we will add them for all components holistically. But good spot 👍

**Pull request merged**